src/main/java/org/wso2/carbon/apimgt/securityenforcer/PingAISecurityHandler.java

```diff
@@ -0,0 +1,328 @@
+/*
+ *  Copyright WSO2 Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wso2.carbon.apimgt.securityenforcer;
+
+import org.apache.axis2.AxisFault;
+import org.apache.axis2.Constants;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpStatus;
+import org.apache.synapse.Mediator;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseConstants;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.rest.AbstractHandler;
+import org.apache.synapse.rest.RESTConstants;
+import org.apache.synapse.transport.passthru.PassThroughConstants;
+import org.apache.synapse.transport.passthru.util.RelayUtils;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.wso2.carbon.apimgt.gateway.handlers.Utils;
+import org.wso2.carbon.apimgt.gateway.handlers.security.AuthenticationContext;
+import org.wso2.carbon.apimgt.securityenforcer.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.securityenforcer.utils.AISecurityException;
+import org.wso2.carbon.apimgt.securityenforcer.utils.AISecurityHandlerConstants;
+import org.wso2.carbon.apimgt.securityenforcer.utils.SecurityUtils;
+
+import java.util.Date;
+import java.util.TreeMap;
+
+/**
+ * This class is Handling the Ping Identity analysis. This class will use inside each API as securityenforcer.
+ * It will fetch some of meta data from incoming message and send them to PING API Security Enforcer.
+ */
+public class PingAISecurityHandler extends AbstractHandler {
+
+    private static final Log log = LogFactory.getLog(PingAISecurityHandler.class);
+
+    public PingAISecurityHandler() {
+        log.debug("Ping AI Security Handler initialized");
+    }
+
+    /**
+     * This method will handle the request. For every request gateway receives, this is method will invoke first for this handler
```

Comment:
please fix all places where it goes beyond 120 limit.

Reply:
fixed with https://github.com/wso2-extensions/apim-handler-pingai/pull/22